core/src/main/java/org/elasticsearch/index/query/RegexpQueryParser.java

```diff
@@ -72,11 +61,11 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                         currentFieldName = parser.currentName();
                     } else {
                         if ("value".equals(currentFieldName)) {
-                            value = parser.objectBytes();
+                            value = parser.text();
```

Comment:
Thanks for your answer. Yes that makes it clear. So we should consider this a bug in the java API and fix it.